lib/sparrow/h2_client_adapter/chatterbox.ex

```diff
@@ -35,6 +37,9 @@ defmodule Sparrow.H2ClientAdapter.Chatterbox do
             "action=open_conn, response=#{inspect({:ok, connection_ref})}"
           end)
 
+        Process.unlink(connection_ref)
```

Comment:
We should have a test that fails without this change, so that we don't drop it by mistake. In this test we should mock `H2Adapter`, `start_link` process in it and crash it, then assert on worker's status.